src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -12,21 +12,54 @@
 </template>
 
 <script lang="ts">
-import { mapGetters } from 'vuex'
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Vue, Watch } from 'vue-property-decorator'
 import Block from '@/components/interaction-designer/Block.vue'
+import { namespace } from 'vuex-class'
+import {IBlock, IFlow} from '@floip/flow-runner'
+import { find, isEqual, cloneDeep } from 'lodash'
+import { IValidationStatus } from '@/store/validation'
 
-  @Component({
-    components: {
-      Block,
-    },
+const flowVuexNamespace = namespace('flow')
+const validationVuexNamespace = namespace('validation')
 
-    computed: {
-      ...mapGetters('flow', ['activeFlow']),
-    },
-
-  })
+@Component({
+  components: {
+    Block,
+  },
+})
 export default class BuilderCanvas extends Vue {
+  @Watch('activeFlow', { deep: true, immediate: true })
+  async onActiveFlowChanged(newFlow: IFlow) {
+    console.debug('active flow has changed, validating ...', newFlow)
+    await this.validate_flow({ flow: newFlow })
+  }
+
+  @Watch('blocksOnActiveFlowForWatcher', { deep: true, immediate: true })
```

Comment:
Would we be able to avoid the indirection of the computed getter (`blocksOnActiveFlowForWatcher`) by using  `activeFlow.blocks` as the string here? It looks like vue supports nesting like that via: https://vuejs.org/v2/api/#vm-watch

I think I'm misunderstanding how come we need the computed getter and cloning :) Would you be willing to help me understand?

Reply:
I'm thinking about performance— We're adding overhead with validation, so we'll need to make sure we do some profiling and testing with larger trees after this feature is done to test things out :)

Reply:
@bzabos : 
We want to watch a state here (activeFlow.blocks), but in addition we want to get correct `old` & `new` new values for comparison.
If I watch only `activeFlow.blocks`, I couldn't have the old & new values correctly. They always have the same value 😞  . The only approach I found so far is to put the state under a getter & make cloneDeep 💡 . (check this: https://stackoverflow.com/questions/50523609/how-to-retrieve-the-old-and-new-values-of-a-store-state)
I would be happy if you have another idea to avoid this approach 😄 

For now, there are 02 solutions:
- solution1: comparing new & old values, we watch the getter `blocksOnActiveFlowForWatcher` which makes `cloneDeep` of `activeFlow.blocks`
- solution2: Without comparing new & old, we just need to watch over `activeFlow.blocks`

I tried to make simple profiling in the watcher
```
  @Watch('...', { deep: true, immediate: true })
  async onBlocksInActiveFlowChanged(newBlocks: IBlock[], oldBlocks: IBlock[]) {
    const t0 = performance.now()
      ...
    const t1 = performance.now()
    console.log("Profiling ", t1 - t0, 'ms')
  }
```

The result show us the solution1 has a significant advantage, check profiling result here https://docs.google.com/document/d/1xwMCxm0ilFExehaNMq8SmvrTuVFZvaHJVwrttozVHIE/edit?usp=sharing